### PR TITLE
Replace JSX with Jason

### PR DESCRIPTION
**Pull request by AndrewDryga**
Jason is the new de-facto library that is used in Ecto and Phoenix,
ExCoveralls was the only library that added JSX dependency. 

Jason is blazing fast too.

### CODE REVIEW

File: lib/excoveralls/circle.ex [hunk 1]
```diff
@@ -7,14 +7,14 @@ defmodule ExCoveralls.Circle do
   def execute(stats, options) do
     json = generate_json(stats, Enum.into(options, %{}))
     if options[:verbose] do
-      IO.puts JSX.prettify!(json)
```
Comment (AndrewDryga) [inline lib/excoveralls/circle.ex]: Only thing that we drop then is pretty printing which is not that bad. We can get it back after Jason implements it https://github.com/michalmuskala/jason/issues/15

File: lib/excoveralls/settings.ex [hunk 2]
```diff
@@ -60,7 +60,7 @@ defmodule ExCoveralls.Settings do
 
   defp read_config_file(file_name) do
     if File.exists?(file_name) do
-      case File.read!(file_name) |> JSX.decode do
+      case File.read!(file_name) |> Jason.decode do
         {:ok, config} -> Enum.into(config, Map.new)
```
Comment (michalmuskala) [inline lib/excoveralls/settings.ex]: Jason will always return a map

File: test/json_test.exs [hunk 3]
```diff
@@ -45,10 +45,9 @@ defmodule ExCoveralls.JsonTest do
       Json.execute(@source_info)
     end) =~ @stats_result
 
-    assert(File.read!(report) =~ ~s({"source_files":[{"name":"test/fixtures/test.ex","source":"defmodule Test do\\n  def test do\\n  end\\nend\\n","coverage":[0,1,null,null]}]}))
+    assert(File.read!(report) =~ ~s({"source_files":[{"coverage":[0,1,null,null],"name":"test/fixtures/test.ex","source":"defmodule Test do\\n  def test do\\n  end\\nend\\n"}]}))
```
Comment (AndrewDryga) [inline test/json_test.exs]: Jason orders JSON a little bit differently, data in JSON is not changed.

**Comment by parroty** [comment]
Thanks for the updates!

**Pull request merged**